cmake/SetupBOUTThirdParty.cmake

```
@@ -297,7 +297,11 @@ if (BOUT_USE_SUNDIALS)
     FetchContent_MakeAvailable(sundials)
     message(STATUS "SUNDIALS done configuring")
   else()
-    find_package(SUNDIALS 4 REQUIRED)
+    ENABLE_LANGUAGE(C)
+    find_package(SUNDIALS REQUIRED)
```

I take it:

```suggestion
    find_package(SUNDIALS 4.0.0 REQUIRED)
```

doesn't work?

---

No, that only seems to work for 4.x.x. For a newer version, I get a warning like
```
CMake Warning at cmake/FindSUNDIALS.cmake:34 (find_package):
  Could not find a configuration file for package "SUNDIALS" that is
  compatible with requested version "4".

  The following configuration files were considered but not accepted:

    .../SUNDIALSConfig.cmake, version: 7.0.0

Call Stack (most recent call first):
  cmake/SetupBOUTThirdParty.cmake:301 (find_package)
  CMakeLists.txt:479 (include)
```
from [here](https://github.com/boutproject/BOUT-dev/blob/ed69fd674bef813f01656bd99962fad880b922f9/cmake/FindSUNDIALS.cmake#L34) (with QUIET removed). This leads to a linking error later.

With cmake 3.19, I think `find_package(SUNDIALS 4...7 REQUIRED)` might work.

---

Oh yes, pretty sure `4...7` is the correct syntax

---

Turns out that the `4...7` syntax doesn't work with
https://github.com/LLNL/sundials/blob/b685654c6958699821fc68a822b66076fec8cae7/CMakeLists.txt#L267
so the current manual check is probably the best option.

---

Ah, yes, I see now, that's very unfortunate. `SameMajorVersion` means "the requested version must have exactly the same major version", and makes no allowances for requesting "any of these major versions".

Agreed, the manual check is the best option.